File: Net5.AspNet.Workshop.Workshop.Api/Infrastructure/FilterParameters/EnrollmentFilter.cs
using Net5.AspNet.Workshop.Infrastructure.Dtos;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Net5.AspNet.Workshop.Workshop.Api.Infrastructure.FilterParameters
{
    public class EnrollmentFilter
    {
        public EnrollmentFilter()
        {
            EnrollmentStatesIds = new List<int>();            
        }
        public string WorshopTitle { get; set; }
        public string EnrolledName { get; set; }
        public List<int> EnrollmentStatesIds { get; set; }
        public DateTime? StarDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int WorkshopId { get; set; }
        public int EnrolledId { get; set; }

        public string EnrolledIdentificationNumber { get; set; }
        public WorkshopDto Workshop { get; set; }
    }
}

File: Net5.AspNet.Workshop.Client/ClientApp/src/app/services/data/workshop/enrollment-filter.ts
import { Workshop } from "../../../models/workshop";

export class EnrollmentFilter {
  constructor() {
    this.EnrolledName = "";
    this.EnrolledIdentificationNumber = "";
    this.WorshopTitle = "";
    this.EnrollmentStatesIds = new Array<number>();
    this.WorkshopId = 0;
    this.EnrolledId = 0;
    this.Workshop = new Workshop();
  }

  public EnrolledName: string;
  public WorshopTitle: string;
  public EnrollmentStatesIds: Array<number>;
  public StartDate?: Date;
  public EndDate?: Date;

  
  public EnrolledIdentificationNumber: string;
  public EnrolledId: number;
  public WorkshopId: number;
  public Workshop: Workshop;

}
